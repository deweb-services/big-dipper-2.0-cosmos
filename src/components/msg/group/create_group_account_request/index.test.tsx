import React from 'react';
import { RecoilRoot } from 'recoil';
import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import { MsgCreateGroupAccountRequest } from '@models';
import CreateGroupAccountRequest from '.';

// ==================================
// mocks
// ==================================

jest.mock('@components', () => ({
  Name: (props) => <div id="Name" {...props} />,
}));

jest.mock('next-translate/Trans', () => (
  (props) => <div id="Trans" {...props} />
));

// ==================================
// unit tests
// ==================================
describe('screen: TransactionDetails/Grant', () => {
  it('matches snapshot', () => {
    const message = new MsgCreateGroupAccountRequest({
      category: 'group',
      type: 'MsgCreateGroupAccountRequest',
      admin: 'admin',
    });

    const component = renderer.create(
      <RecoilRoot>
        <MockTheme>
          <CreateGroupAccountRequest
            message={message}
          />
        </MockTheme>
      </RecoilRoot>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    expect(component.root.findByProps({ id: 'Trans' }).props.i18nKey).toEqual('message_contents:MsgCreateGroupAccountRequest');
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
