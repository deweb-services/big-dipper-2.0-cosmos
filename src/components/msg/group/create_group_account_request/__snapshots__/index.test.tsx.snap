// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: TransactionDetails/Grant matches snapshot 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1"
>
  <div
    components={
      Array [
        <Name
          address="admin"
          name="admin"
        />,
      ]
    }
    i18nKey="message_contents:MsgCreateGroupAccountRequest"
    id="Trans"
  />
</p>
`;
